taskana-adapter-camunda-outbox-rest/src/main/java/pro/taskana/adapter/camunda/outbox/rest/service/CamundaTaskEventsService.java

```diff
@@ -0,0 +1,252 @@
+        }
+
+    }
+
+    private List<CamundaTaskEventResource> getCreateEvents() {
+
+        List<CamundaTaskEventResource> camundaTaskEventResources = new ArrayList<>();
+
+        try (
+                Connection connection = getConnection();
+                ResultSet createEventsResultSet = prepareAndExecuteCreateEventsQuery(connection)
+        ) {
+
+            camundaTaskEventResources = getCamundaTaskEventResources(createEventsResultSet);
+
+        } catch (SQLException e) {
+            LOGGER.warn("Caught {} while trying to retrieve create events from the outbox", e);
+        } catch (NullPointerException e) {
+            LOGGER.warn("Caught {} while trying to retrieve create events from the outbox", e);
+        }
+        return camundaTaskEventResources;
+    }
+
+
+    private String preparePlaceHolders(int length) {
+        return String.join(",", Collections.nCopies(length, "?"));
+
+    }
+
+    private void setPreparedStatementValues(PreparedStatement preparedStatement, List<Integer> ids) throws SQLException {
+        for (int i = 0; i < ids.size(); i++) {
+            preparedStatement.setObject(i + 1, ids.get(i));
+        }
+    }
+
+
+    private List<CamundaTaskEventResource> getCamundaTaskEventResources(ResultSet createEventsResultSet) throws SQLException, NullPointerException {
+
+        List<CamundaTaskEventResource> camundaTaskEventResources = new ArrayList<>();
+
+        while (createEventsResultSet.next()) {
+
+            CamundaTaskEventResource camundaTaskEventResource = new CamundaTaskEventResource();
+
+            camundaTaskEventResource.setId(createEventsResultSet.getInt(1));
+            camundaTaskEventResource.setType(createEventsResultSet.getString(2));
+            camundaTaskEventResource.setCreated(formatDate(createEventsResultSet.getTimestamp(3)));
+            camundaTaskEventResource.setPayload(createEventsResultSet.getString(4));
+
+            camundaTaskEventResources.add(camundaTaskEventResource);
+
+        }
+
+        return camundaTaskEventResources;
+
+    }
+
+
+    private List<Integer> getIdsAsIntegers(String idsAsJsonArray) {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        List<Integer> idsAsIntegers = new ArrayList<Integer>();
+
+        try {
+            JsonNode idsAsJsonArrayNode = objectMapper.readTree(idsAsJsonArray).get("taskCreationIds");
+
+            if (idsAsJsonArrayNode != null) {
+                idsAsJsonArrayNode.forEach(id -> idsAsIntegers.add(id.asInt()));
+            }
+
+        } catch (IOException e) {
+            LOGGER.warn("Caught {} while trying to read the passed JSON-Object in the POST-Request to delete events from the outbox table", e);
+        }
+        return idsAsIntegers;
+    }
+
+
+    private ResultSet prepareAndExecuteCreateEventsQuery(Connection connection) throws SQLException {
+
+        PreparedStatement preparedStatement = connection.prepareStatement(SQL_GET_CREATE_EVENTS);
```

Comment:
use a try-with-resources for this preparedStatement in order to close it correctly

Reply:
changed